pi-blaster.c

```diff
@@ -57,17 +59,38 @@ static uint8_t known_pins[] = {
 		23,     // P1-16
 		24,     // P1-18
 		25,     // P1-22
-		10,     // P1-19
-		9,      // P1-21
-		11,     // P1-23
 };
 
-// pin2gpio array is not setup as empty to avoid locking all GPIO
-// inputs as PWM, they are set on the fly by the pin param passed.
-static uint8_t pin2gpio[16];
+// Create a list of reserved GPIO pins
+// http://elinux.org/RPi_Low-level_peripherals
+static uint8_t banned_pins[MAX_CHANNELS] = {
+		6,      	// On Model B, it is in use for the Ethernet function
+		28,		// board ID and are connected to resistors R3 to R10 (only on Rev1.0 boards).
+		29,		// board ID and are connected to resistors R3 to R10 (only on Rev1.0 boards).
+		30,		// board ID and are connected to resistors R3 to R10 (only on Rev1.0 boards).
+		31,		// board ID and are connected to resistors R3 to R10 (only on Rev1.0 boards).
+//		40,		// used by analogue audio
+//		45,		// used by analogue audio
+		46,		// HDMI hotplug detect
+		47,		// 47 to 53 are used by the SD card interface.
+		48,		// 47 to 53 are used by the SD card interface.
+		49,		// 47 to 53 are used by the SD card interface.
+		50,		// 47 to 53 are used by the SD card interface.
+		51,		// 47 to 53 are used by the SD card interface.
+		52,		// 47 to 53 are used by the SD card interface.
+		53,		// 47 to 53 are used by the SD card interface.
+};
 
 // Set num of possible PWM channels based on the known pins size.
-#define NUM_CHANNELS    (sizeof(known_pins)/sizeof(known_pins[0]))
+
+//#define NUM_CHANNELS    (sizeof(known_pins)/sizeof(known_pins[0]))
+uint8_t NUM_CHANNELS = (sizeof(known_pins)/sizeof(known_pins[0]));
```

Comment:
This is not a constant anymore so you should rename it to be lower case: `num_channels`.